src/scalems/radical/raptor.py

```diff
@@ -226,33 +188,42 @@ def __init__(self, configuration: Configuration):
                 )
         super(ScaleMSMaster, self).__init__()
 
-    def result_cb(self, requests: typing.Sequence[Request]):
+    def result_cb(self, requests: typing.Sequence[rp.TaskDescription]):
```

Comment:
requests: dict with a `description` field representing a TaskDescription.
@andre-merzky will follow up.

Reply:
Should we assume that elements of *requests* are effectively of the return type of `Task.as_dict()`?
https://github.com/radical-cybertools/radical.pilot/blob/206e104b0d23d489603fd46e5572be580b92b669/src/radical/pilot/task.py#L178

Reply:
@andre-merzky note: ‘exception’ field is a tuple: type name, exception message

Reply:
I seem to have mixed up some notes. @andre-merzky can you please clarify?

Results of the function call (for the task executor) are added as the {*val* element | `'val'` field | 4th element} of the { named tuple | dictionary | simple sequence object } in the `'result'` field of a *request*.

Reply:
You probably want `worker_mpi` by now.  There indeed the returned data are `out, err, ret, val, exc` with `exc` being a tuple.

Reply:
* `MPIWorker` (from `radical.pilot.raptor.worker_mpi`)
* `request` schema currently depends on the type of worker, but this will be normalized soon

`result_cb` signature receives a sequence of dictionary representations of Tasks (schema to be formalized soon).

These distionaries will have _each_ of _out_, _err_, _ret_, _val_, and _exc_. (`result`) field no longer exists.